packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -435,8 +473,8 @@ class RenderGrid extends RenderVirtualViewport<GridParentData> {
       double tileTop = _specification.rowOffsets[placement.row] + gridTopPadding;
       double tileBottom = _specification.rowOffsets[placement.row + placement.rowSpan] + gridTopPadding;
 
-      double childWidth = tileRight - tileLeft - placement.padding.horizontal;
-      double childHeight = tileBottom - tileTop - placement.padding.vertical;
+      double childWidth = math.max(0.0, tileRight - tileLeft - _specification.columnSpacing);
```

Comment:
i would have expected half the columnSpacing to be applied to each column's width unless there's only one column in which case it would not be applied.